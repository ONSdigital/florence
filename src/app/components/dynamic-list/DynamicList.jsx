import React from "react";
import DynamicListItem from "./DynamicListItem";
import PropTypes from "prop-types";
import Input from "../Input";

const propTypes = {
    title: PropTypes.string.isRequired,
    handleSearchInput: PropTypes.func,
    listItems: PropTypes.arrayOf(
        PropTypes.shape(DynamicListItem.propTypes)).isRequired
}

const DynamicList = props => {
    return (
        <div>
            <span className="dynamic-list__title">{props.title}</span>
            {props.handleSearchInput && <Input id="search-content-types" placeholder="Search viewers by name or email" onChange={props.handleSearchInput} />}
            <ul>
                {props.listItems.map((row, index)=>{ return <DynamicListItem key={`dynamicListRow-${index}`}{...row}/>})}
            </ul>
        </div>
    )
};

DynamicList.propTypes = propTypes;
export default DynamicList;
