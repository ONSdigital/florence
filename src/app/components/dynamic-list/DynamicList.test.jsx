import React from "react";
import DynamicList from "./DynamicList";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

describe("DynamicList", () => {
    const filteredList = [
        {
            title: "Simon Davies",
            desc: " Simon.davies@ons.gov.uk",
            icon: "Person",
            iconName: "Viewer",
            buttonName: "Add",
            buttonCallback: () => {},
        },
        {
            title: "Michael Miller",
            desc: " Michael.miller@ons.gov.uk",
            icon: "Person",
            iconName: "Viewer",
            buttonName: "Add",
            buttonCallback: () => {},
        },
    ];
    describe("given props does it create a list", () => {
        const dynamicListProps = {
            title: "Add member to team",
            listItems: filteredList,
            listHeightClass: "fifthHeight",
            noResultsText: "No viewers found",
            filterPlaceholder: "Search viewers by name or email",
            headingLevel: "1",
            handleSearchInput: () => {},
        };
        const component = mount(<DynamicList {...dynamicListProps} />);
        it("should display a dynamic list", () => {
            expect(component.find(".dynamic-list__title").exists()).toBe(true);
            expect(component.find(".dynamic-list-item").length).toBe(2);
            expect(component.find(".dynamic-list-item__btn").length).toBe(2);
        });
        it("matches the snapshot", () => {
            const componentForSnapshot = renderer.create(<DynamicList {...dynamicListProps} />);
            expect(componentForSnapshot.toJSON()).toMatchSnapshot();
        });
    });
});
