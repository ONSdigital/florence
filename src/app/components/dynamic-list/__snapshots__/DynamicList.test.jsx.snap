// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicList given props does it create a list matches the snapshot 1`] = `
<div>
  <h1
    className="dynamic-list__title"
    data-testid="dynamic-list-title"
  >
    Add member to team
  </h1>
  <label
    className="visually-hidden"
    htmlFor="add-member-to-team-search-content-types"
  >
    Search viewers by name or email
  </label>
  <div
    className="form__input"
  >
    <label
      className="form__label"
      htmlFor="add-member-to-team-search-content-types"
    />
    
    <input
      autoComplete="new-password"
      autoFocus={false}
      className="input"
      disabled={false}
      id="add-member-to-team-search-content-types"
      name="add-member-to-team-search-content-types"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search viewers by name or email"
      type="text"
    />
    
  </div>
  <ul
    className="dynamic-list__container fifthHeight"
    data-testid="dynamic-list-container"
  >
    <li
      className="dynamic-list-item"
    >
      <svg
        aria-label="Viewer"
        className="dynamic-list-item__icon"
        fill="none"
        role="img"
        viewBox="0 0 15 15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.667 7.5a3.332 3.332 0 1 0 0-6.667 3.332 3.332 0 1 0 0 6.667Zm0 1.667C4.442 9.167 0 10.283 0 12.5v1.667h13.333V12.5c0-2.217-4.441-3.333-6.666-3.333Z"
          fill="#000000"
        />
      </svg>
      <span
        className="dynamic-list-item__title"
      >
        Simon Davies
      </span>
      <span
        aria-hidden="true"
        className="dynamic-list-item__separator"
      >
         
        •
         
      </span>
      <span
        className="dynamic-list-item__desc"
      >
         Simon.davies@ons.gov.uk
      </span>
      <button
        aria-label="Add Simon Davies"
        className="dynamic-list-item__btn"
        onClick={[Function]}
        type="button"
      >
        Add
      </button>
    </li>
    <li
      className="dynamic-list-item"
    >
      <svg
        aria-label="Viewer"
        className="dynamic-list-item__icon"
        fill="none"
        role="img"
        viewBox="0 0 15 15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.667 7.5a3.332 3.332 0 1 0 0-6.667 3.332 3.332 0 1 0 0 6.667Zm0 1.667C4.442 9.167 0 10.283 0 12.5v1.667h13.333V12.5c0-2.217-4.441-3.333-6.666-3.333Z"
          fill="#000000"
        />
      </svg>
      <span
        className="dynamic-list-item__title"
      >
        Michael Miller
      </span>
      <span
        aria-hidden="true"
        className="dynamic-list-item__separator"
      >
         
        •
         
      </span>
      <span
        className="dynamic-list-item__desc"
      >
         Michael.miller@ons.gov.uk
      </span>
      <button
        aria-label="Add Michael Miller"
        className="dynamic-list-item__btn"
        onClick={[Function]}
        type="button"
      >
        Add
      </button>
    </li>
  </ul>
</div>
`;
